msteams-platform/toolkit/provision.md

```diff
@@ -1,616 +1,436 @@
 
-If the predefined templates don't meet your app requirements, you can customize the ARM templates under `templates\azure` folder. For example, you can customize the ARM template to create some extra Azure resources for your app. You need to have basic knowledge of bicep language, which is used to author ARM template.
+ ```typescript
+ writeToEnvironmentFile:
+   clientId: AAD_APP_CLIENT_ID
+   clientSecret: SECRET_AAD_APP_CLIENT_SECRET
+   objectId: AAD_APP_OBJECT_ID
+   tenantId: AAD_APP_TENANT_ID
+   authority: AAD_APP_OAUTH_AUTHORITY
+   authorityHost: AAD_APP_OAUTH_AUTHORITY_HOST
+ ```
 
-To ensure the TeamsFx tool functions properly, customize ARM template that satisfies the following requirements:
+1. Add values for each environment variable from step 2.
 
-* Ensure that the folder structure and file name remain unchanged. The tool may append new content to the existing files when you add more resources or capabilities to your project.
-* Ensure that the name of auto-generated parameters and its property names remain unhanged. The auto-generated parameters may be used when you add more resources or capabilities to your project.
-* Ensure that the output of auto-generated ARM template is unchanged. You can add more outputs to ARM template. The output is `.fx\states\state.{env}.json` and can be used in other features, such as deploy and validate manifest files.
+    1. Add below environment variables and their values to `env\.env.{env}` file.
 
-### Customize Teams app
+    ```typescript
+    AAD_APP_CLIENT_ID=<value of Azure AD application's client id (application id)> # example: 00000000-0000-0000-0000-000000000000
+    AAD_APP_OBJECT_ID=<value of Azure AD application's object id> # example: 00000000-0000-0000-0000-000000000000
+    AAD_APP_TENANT_ID=<value of Azure AD's Directory (tenant) id>> # example: 00000000-0000-0000-0000-000000000000
+    AAD_APP_OAUTH_AUTHORITY=<value of Azure AD's authority> # example: https://login.microsoftonline.com/<Directory (tenant) ID>
+    AAD_APP_OAUTH_AUTHORITY_HOST=<host of Azure AD's authority> # example: https://login.microsoftonline.com
+    AAD_APP_ACCESS_AS_USER_PERMISSION_ID=<id of access_as_user permission> # example: 00000000-0000-0000-0000-000000000000
+    ```
 
-You can customize your bot or the Teams app by adding configuration snippets to use an Azure AD app created for your Teams app.
-Perform in the following ways to customize the Teams app:
+    1. If your application requires an Azure AD app client secret, add below environment variable and its value to `env\.env.{env}.user` file.
 
-* [Use an existing Azure AD app for your Teams app](#use-an-existing-azure-ad-app-for-your-teams-app)
-* [Use an existing Azure AD app for your bot](#use-an-existing-azure-ad-app-for-your-bot)
+    ```typescript
```

Comment:
This is the codesnippet for .env file. AFAIK, markdown does not have syntax highlighting for .env file, so may delete the language part.